### DISCUSSION TITLE
single examples not showing up in latest Swagger UI

### DISCUSSION THREAD

**Discussion by havok2063** [fastapi/fastapi#10537] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Dict
from fastapi import Depends, FastAPI, Query

app = FastAPI(title='test', description='test', version='0.1.0',
              dependencies=[])

@app.get("/", summary='Hello World route', response_model=Dict[str, str])
def hello():
    return {"Hello": "This is the FastAPI World"}

@app.get("/test", summary='test route',
                response_model=dict, response_model_exclude_unset=True)
def test(a: int = Query(..., description='An integer', examples=[5]),
         b: str = Query(..., description='old deprecated example', example='old_example'),
         c: int = Query(..., description='openapi example', openapi_examples={'default':{'value':5}})):
    """ test """
    return {'a': a, 'b': b, 'c': c}


if __name__ == "__main__":
    app.run()
```

### Description

I mostly use single item examples in my Query, Path, Body parameters.  Using `examples` does not render correctly in the UI.  The old singular `example`, still renders correctly, but I know that will be deprecated soon.  If I switch to the `openapi_examples`, it does render correctly but this seems better suited for multiple examples, and would add a lot of boilerplate to the code in order to specify the correct format just for a single value example.  

I've read through this page, https://fastapi.tiangolo.com/tutorial/schema-extra-example/#technical-details and it seems like as of Swagger 5.0, the use of the new `examples` in OpenAPI 3.1.0 should render in the UI successfully.  Did I misinterpret that?  If not, could there be a bug, or if so, am I stuck with using the `openapi_examples` syntax for the time being?  I'd like to move away from the deprecated `example` syntax to future proof my code.  

I've confirmed I'm using OpenAPI 3.1.0 and Swagger 5+:
`'{"swaggerUi":{"version":"5.9.1","gitRevision":"g2d807965","gitDirty":true,"buildTimestamp":"Wed, 25 Oct 2023 08:20:31 GMT"}}' `

### Operating System

macOS

### FastAPI Version

0.104.0

### Pydantic Version

2.4.2

### Python Version

Python 3.10.12

**Comment by hhartzer** [comment]
I'm seeing this as well. If I use `examples`, it does not show up in Swagger for Query/Path. If I use `example`, it does.

**Comment by atillaguzel** [comment]
I confirm I have the same issue with FastAPI version `0.104.1`

**Comment by HaKuNamaTT-ZZH** [comment]
Hi, I am using 0.108.0 now, this weird thing has appeared a long time in the project I work.
I hope you guys could FIX this, or UPDATE the documentation, otherwise it is misleading.

Clearly, it is also a boilerplate to use 'json_schema_extra' to define examples.

**Comment by ajayrawatsap** [comment]
Same issue for me in  0.103.1.

It is a bug and I think it should be labeled as a bug and not a question.

**Reply by hhartzer** [reply]
@tiangolo what is the process to have this looked at? Since this appears to be well confirmed as a bug, is there someone we can tag to have this converted into a bug?

Thank you!

**Comment by havok2063** [comment]
Is there a way to change this to a bug or migrate it?

**Comment by andreaevooq** [comment]
I am experiencing the same issue

**Comment by admo1** [comment]
Same issue, any update ?

**Comment by GodMoonGoodman** [comment]
same issue,

python 3.10.8
fastapi 0.109.2

**Comment by graysonpike** [comment]
Same issue here. I've confirmed that the examples do not show up in the `openapi.json` using either the `examples` or `openapi_examples` syntax. I am specifying them in the `Field` class of my Pydantic model, which is used as the type of a query param.
Python 3.10
FastAPI 0.110.0

**Comment by jonocodes** [comment]
same here, and its not showing in the openapi spec
fastapi 0.111.0

**Comment by ginomempin** [comment]
I am on FastAPI **0.111.0** and I can confirm that it _does_ work now with `openapi_examples`, as explained here:  
https://fastapi.tiangolo.com/tutorial/schema-extra-example/#using-the-openapi_examples-parameter

### Code:
```
@router.get("/test1")
def test(
    one: Annotated[
        int,
        Query(
            openapi_examples={
                "example1": {
                    "summary": "Example 1",
                    "description": "Example 1",
                    "value": 5,
                }
            }
        ),
    ],
    two: Annotated[
        str,
        Query(
            openapi_examples={
                "example1": {
                    "summary": "Example 1",
                    "description": "Example 1",
                    "value": "value1",
                },
                "example2": {
                    "summary": "Example 2",
                    "description": "Example 2",
                    "value": "value2",
                },
            }
        ),
    ],
    three: Annotated[
        dict,
        Body(
            openapi_examples={
                "example1": {
                    "summary": "Example 1",
                    "description": "Example 1",
                    "value": {
                        "key1": "value1",
                        "key2": "value2",
                    },
                },
                "example2": {
                    "summary": "Example 2",
                    "description": "Example 2",
                    "value": {
                        "key3": "value3",
                        "key4": "value4",
                    },
                },
            }
        ),
    ],
) -> dict:
    return {"one": one, "two": two, "three": three}
```

### Rendered Docs:

Query Params:
<img width="522" alt="Screen Shot 2024-06-05 at 19 21 06" src="https://github.com/tiangolo/fastapi/assets/6091865/7b13844d-e646-432f-a37e-067b407f1d91">

Request Body:
<img width="422" alt="Screen Shot 2024-06-05 at 19 25 07" src="https://github.com/tiangolo/fastapi/assets/6091865/c177ca27-1de4-4b2d-a545-50fd754ff80c">

As shown, it works for both single-examples and multi-examples.
I used `Query` in that example, but as mentioned in the docs, it works for `Path` or `Body`.

**Comment by rganowski** [comment]
Scalar `example` should not be marked as deprecated in the FastAPI's  `Query` and `Path`: 

```python
   example: Annotated[
        Optional[Any],
        deprecated(
            "Deprecated in OpenAPI 3.1.0 that now uses JSON Schema 2020-12, "
            "although still supported. Use examples instead."
        ),
    ] = _Unset,
```

because OpenAPI 3.1.0 states for the **Parameter Object**:

<img src="https://github.com/user-attachments/assets/49fdf418-a3d5-4b5e-9d30-d24db9eae308" width="70%">

A deprecation note is attached to the `example` field of the **Schema Object**

<img src="https://github.com/user-attachments/assets/4fd1c5d5-1958-4414-93dd-b4021b80b155" width="70%">

When you put old `example` into `Query` it is used by the FastAPI as a direct attribute of parameter definition in OpenAPI. If you put `examples` FastAPI places it into `schema` of parameter definition. 

1. both, `example` and `examples`, should end up in parameter definition, and only then those are presented in swagger ui
2. but only one of them is legal at a time
3. multiple examples, if applied, should be defined as a dictionary of **Example objects**, not just a list of scalars. As in `openapi_examples`, but then one of `examples` or `openapi_examples` is redundant.
 
Correct me if I'm wrong

**Comment by adriamontoto** [comment]
Same issue here! I'm using 0.115.5.

**Reply by ahemwe** [reply]
Thanks for this discussion. Souldn't this be converted into an issue?

I'm facing the same (`example` works, but raises an Deprecation Warning, while `examples` does not work and `openapi_examples` is not my usecase) issue.

**Reply by gelezo43** [reply]
Have same problem, described in another discussion: https://github.com/fastapi/fastapi/discussions/12710#discussioncomment-11578130

**Comment by YuriiMotov** [comment]
As I see it, the `examples` parameter should be either deprecated or become an alias to `openapi_examples`.

There is no `examples` parameter in openapi 3.1.0/3.1.1 specs for the parameter object that can accept an array of examples.
According to spec, `examples` for the parameter object can only accept `Map[ string, Example Object | Reference Object]` and should be placed outside of `schema`. And that's how `openapi_examples` works.

https://spec.openapis.org/oas/v3.1.0.html#parameter-object

So, use either single `example` or `openapi_examples`

**Comment by dotmitsu** [comment]
I have the same problem in fastapi==0.128.0 and Pydantic pydantic==2.12.5
